File: Programmers/Hash/위장.js
function solution(clothes) {
    let answer = 1;
    let m = new Map();
    
    clothes.map((el,idx)=>{
        if(m.has(el[1])){
            m.set(el[1],m.get(el[1])+1);
        }
        else {
            m.set(el[1],1);
        }
    })
    // for(const [key, value] of m){
    //     answer *= (value+1);
    // }
    m.forEach((value, key)=>{
        answer *= (value+1)
    })
    return answer-1;
}

File: Programmers/Hash/위장.cpp
#include <string>
#include <vector>
#include <iostream>
#include <map>
using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map <string,int> m;
    for(int i = 0 ; i < clothes.size();i++){
           m[clothes[i][1]]++;
    }
    for(auto itr:m){
        answer *= itr.second+1;
    }
    return answer-1;
}
